### ISSUE TITLE
Error in if (any(date_ranges2$start_first_occasion > date_ranges2$end_last_occasion)) { :    missing value where TRUE/FALSE needed

### ISSUE THREAD

**Issue by jimcasaer** [inbo/camtraptor#110]
when using the code below I still get this error : 

Error in if (any(date_ranges2$start_first_occasion > date_ranges2$end_last_occasion)) { : 
  missing value where TRUE/FALSE needed

start_date <- lubridate::as_datetime("2020-4-01")
end_date <- lubridate::as_datetime("2020-9-30")

cam_op <- get_cam_op(GMU8, pred_and(pred_gte("start", start_date), pred_lte("end", end_date)),
  station_col = "locationName",
  use_prefix = FALSE
)

rec_table <- get_record_table(GMU8, pred_and(pred_gte("start", start_date), pred_lte("end", end_date)),
  stationCol = "locationName",
  exclude = NULL,
  minDeltaTime = 0,
  deltaTimeComparedTo = NULL
)

Dethist_wb <- detectionHistory(recordTable = rec_table,
                               camOp = cam_op,
                               stationCol = "Station",
                               speciesCol = "Species",
                               recordDateTimeCol = "DateTimeOriginal",
                               recordDateTimeFormat ="%Y-%m-%d %H:%M:%S",
                               species = "Sus scrofa",
                               occasionLength = 1,
                               day1 = "station",
                               datesAsOccasionNames = FALSE,
                               occasionStartTime = 0,
                               includeEffort = FALSE,
                               timeZone = "UTC",
                               writecsv = FALSE)

dethist <- as.data.frame(Dethist_wb$detection_history)
View(dethist)

occupcov <- unmarkedFrameOccu(y = dethist)
null_p <- occu(~1~1, occupcov) # Model without covariates  
backTransform(null_p, 'det') # Backtransformation
backTransform(null_p, 'state')

**Comment by damianooldoni** [comment]
Note for me. 

This error is the consequence of something happpening at an earlier step in the function. See the warning:
```r
Warning messages:
1: In min(which(!is.na(X))) :
  no non-missing arguments to min; returning Inf
2: In max(which(!is.na(X))) :
  no non-missing arguments to max; returning -Inf
```
We get this at line 197: `date_ranges <- do.call(createDateRangeTable, arg.list0)`. Indeed, `date_ranges` contains 65 rows with `rec.min` or `rec.max` column equal to NA. But there is one deployment with NA in all columns: `2020_8_1549`.

**Comment by damianooldoni** [comment]
Something suspect in the camera operation matrix: all columns are equal to NA:
```r
> all(is.na(cam_op[which(rownames(cam_op) == "2020_8_1549"),]))
[1] TRUE
```
Notice that  `2020_8_1549` is a location name and is linked to two deployments:

```r
> GMU8$deployments %>%
+   dplyr::filter(.data$locationName == "2020_8_1549") %>%
+   select(locationName, deploymentID, start, end)
# A tibble: 2 x 4
  locationName deploymentID                         start               end                
  <chr>        <chr>                                <dttm>              <dttm>             
1 2020_8_1549  45339f2a-33ff-4211-9eec-ff20a5f1555d 2020-08-03 11:10:11 2020-08-10 23:00:00
2 2020_8_1549  ba2d06d4-9b67-4b1b-843c-afbf970cbd90 2020-08-11 08:05:22 2020-09-04 06:02:36
```

Further debugging now in `camtraptor::get_cam_op()`.

**Comment by damianooldoni** [comment]
To solve issue #99, we wrote the PR #104 where we converted 0s to NAs internally. But I did still in an early stage, i.e. while calculating the effort at deploymentID level. However, we group the deployments at locationName level after it and we use `rowSums()` to sum the efforts. As shown above, location `2020_8_1549 `  is linked to two  deployments which are not active at the very same moment, so this means that the sum is always of the form: NA + number  or number + NA, which is always converted to NA.
So, the mapping of 0s to NAs by `dplyr::na_if(..., y = 0)` should occur after this sum.
